### ISSUE TITLE
[Bug]: OpenAIAgent as Top Agent with Tool Retriever doesn't work - Multi-Document Agent

### ISSUE THREAD

**Issue by sreeharissh14** [run-llama/llama_index#15515]
### Bug Description

I have a set of agents that are linked to vector and summary tools. I am using Azure OpenAI GPT-4o as LLM with OpenAIAgent. The top agent is not able to use the correct agentic tool during follow up question. 

For eg. I have a agent called tool_hsg_max attached to top agent. When I try to chat with question like "what is the source of fund for hsg max product ?", it is able to call the "tool_hsg_max" and fetch the correct result. However, when I ask a follow up question like "what is the entry age requirement for this product?", the logs show "Tool with name tool_hsg_max not found, please double check.". This only occurs when I don't mention the product name in the question. 

Is this because the tool_retriever is not able to get the context of the question from chat history? 

### Version

0.10.45

### Steps to Reproduce

Follow up question to the top agent causes this issue in OpenAIAgent when using Azure OpenAI GPT-4o.

**Comment by logan-markewich** [comment]
Probably because the tool retriever isn't returning the tool the second time, but the agent can see it in the chat history, so tries to call it anyways

**Comment by sreeharissh14** [comment]
Yes, I have increased the number of similar results to the maximum and solved the issue. But, would like to understand if there is a cleaner approach to solve the problem? For eg, is it possible to pass context to ObjectRetriever? Or are there other utilities for handling this?

**Comment by sreeharissh14** [comment]
Hi @logan-markewich, can you suggest any mechanism to overcome this limitation for ObjectIndex?

**Comment by sreeharissh14** [comment]
Hi @dosu, how do I use CondensePlusContextChatEngine as tool retriever? Give me example code with ObjectIndex and sample tools.
